lib/libutee/tee_api_operations.c

```diff
@@ -1195,9 +1235,24 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle operation,
 	return res;
 }
 
+TEE_Result __GP11_TEE_CipherDoFinal(TEE_OperationHandle operation,
+				    const void *srcData, uint32_t srcLen,
+				    void *destData, uint32_t *destLen)
+{
+	TEE_Result res = TEE_SUCCESS;
+	size_t dl = 0;
+
+	if (destLen)
+		__utee_check_inout_annotation(destLen, sizeof(*destLen));
+	dl = *destLen;
```

Comment:
This relates to the "destLen can be NULL" issue we have also with the TEE_AE* functions.

I propose to treat a NULL destLen as if *destLen was 0, as we're doing today.
Then we'll address the "destLen can be NULL" issue at a later stage. Depending on how it's approached it might not be an easy task. I've already done one halfhearted attempt and it wasn't pretty.

Reply:
I meant:
```c
	if (destLen) {
		__utee_check_inout_annotation(destLen, sizeof(*destLen));
		dl = *destLen;
	}
	res = TEE_CipherDoFinal(operation, srcData, srcLen, destData, &dl);
	if (destLen)
		*destLen = dl;

Reply:
OK, I'll add that.